lsst/ts/cbp/statemachine.py

```diff
@@ -0,0 +1,527 @@
+            self.tel_parked.put(self.parked_topic)
+            if self.model.panic_status == 1:
+                self.fault()
+
+            await asyncio.sleep(self.frequency)
+
+    async def do_moveAltitude(self,id_data):
+        """Moves the altitude axis of the CBP.
+
+        Parameters
+        ----------
+        id_data
+
+        Returns
+        -------
+        None
+
+        """
+        self.assert_enabled("moveAltitude")
+        self.model.move_altitude(id_data.data.altitude)
+        self.cmd_moveAltitude.ackInProgress(id_data, "In progress")
+        await asyncio.sleep(self.cbp_speed*self.factor)
+
+    async def do_setFocus(self, id_data):
+        """Sets the focus.
+
+        Parameters
+        ----------
+        id_data
+
+        Returns
+        -------
+        None
+
+        """
+        self.assert_enabled("setFocus")
+        self.model.change_focus(id_data.data.focus)
+
+    async def do_park(self):
+        """Parks the CBP.
+
+        Returns
+        -------
+        None
+
+        """
+        self.assert_enabled("park")
+        self.model.park()
+
+    async def do_changeMask(self,id_data):
+        """Changes the mask.
+
+        Parameters
+        ----------
+        id_data
+
+        Returns
+        -------
+        None
+
+        """
+        self.assert_enabled("changeMask")
+        self.model.change_mask(id_data.data.mask)
+
+    async def do_clearFault(self, id_data):
+        """
+
+        Parameters
+        ----------
+        id_data
+
+        Returns
+        -------
+        None
+
+        """
+        pass # TODO: finish clearFault
+
+    async def do_enterControl(self, id_data):
+        """
+
+        Parameters
+        ----------
+        id_data
+
+        Returns
+        -------
+        None
+
+        """
+        pass #TODO: finish enterControl
+
+    async def begin_enable(self, id_data):
+        """Overrides the begin_enable function in salobj.BaseCsc to make sure the CBP is unparked.
+
+        Parameters
+        ----------
+        id_data
+
+        Returns
+        -------
+        None
+
+        """
+        self.model._cbp.set_park()
+
+
+class CBPRemote:
+    """This the class that implements the remote functionality for the CBP CSC.
+
+    Attributes
+    ----------
+    remote: salobj.Remote
+        The salobj remote that is used to send commands and receive events and telemetry.
+
+    log: logging.Logger
+        This is the log for the current class.
+
+    """
+    def __init__(self):
+        self.remote = salobj.Remote(SALPY_CBP)
+        self.log = logging.getLogger(__name__)
+
+    async def standby(self):
+        """Calls and awaits the standby command. Logs the acknowledgement code from the command.
+
+        Returns
+        -------
+        None
+
+        """
+        standby_topic = self.remote.cmd_standby.DataType()
+        standby_ack = await self.remote.cmd_standby.start(standby_topic,timeout=10)
```

Comment:
all of these timeouts will need to be moved into a config file in the next version

Reply:
Agreed.